### PR TITLE
Make ConnectionSpec more uniform.

### PR DESCRIPTION
**Pull request by swankjesse**
Closes https://github.com/square/okhttp/issues/1831

### CODE REVIEW

File: okhttp-tests/src/test/java/com/squareup/okhttp/ConnectionSpecTest.java [hunk 1]
```diff
@@ -206,7 +256,47 @@ public void tls_missingTlsVersion() throws Exception {
...
+    assertTrue(set.add(ConnectionSpec.CLEARTEXT));
+    assertTrue(set.add(allTlsVersions));
+    assertTrue(set.add(allCipherSuites));
+
+    assertTrue(set.remove(ConnectionSpec.MODERN_TLS));
+    assertTrue(set.remove(ConnectionSpec.COMPATIBLE_TLS));
+    assertTrue(set.remove(ConnectionSpec.CLEARTEXT));
+    assertTrue(set.remove(allTlsVersions));
+    assertTrue(set.remove(allCipherSuites));
+  }
```
Comment (JakeWharton) [inline okhttp-tests/src/test/java/com/squareup/okhttp/ConnectionSpecTest.java]: Can we assert empty at the end for any future change sanity?

**Comment by JakeWharton** [comment]
LGTM (despite being some of the code I am least familiar with here)

**Comment by swankjesse** [comment]
FYI @nfuller: new APIs, `ConnectionSpec.Builder.allEnabledTlsVersions()` and `ConnectionSpec.Builder.allEnabledCipherSuites` which replace passing null to those methods.

**Comment by nfuller** [comment]
Ack. Thanks.

**Pull request merged**